aws/session/session.go

```diff
@@ -36,7 +36,7 @@ const (
 
 // ErrSharedConfigSourceCollision will be returned if a section contains both
 // source_profile and credential_source
-var ErrSharedConfigSourceCollision = awserr.New(ErrCodeSharedConfig, "only source profile or credential source can be specified, not both", nil)
+var ErrSharedConfigSourceCollision = awserr.New(ErrCodeSharedConfig, "only one credential type may be specified per profile: source profile, credential source, credential process, web identity token, or sso", nil)
```

Comment:
Can we port this error msg to v2 too?  This one is better, as it explicitly states what sources are allowed.

Reply:
Yup this message came from that version: https://github.com/aws/aws-sdk-go-v2/blob/92a8d37a42f85e3eb8b445c126a3b02d17e887b8/config/shared_config.go#L924

Reply:
@skotambkar @skmcgrail Is there a reason this is required that only 1 of these sources can be provided? A common workaround for, well anyone using AWS SSO was to use the credential_process string in their profiles to extract the SSO creds.

With this in place, my reading is that no one can use this version of the Go SDK going forward if they also depend on another SDK that doesn't yet work with AWS SSO credentials natively (without I guess finding another hack for lack of SDK support). The C++ SDK I don't believe supports SSO at all.

Reply:
I agree on this. This error makes sense if there's a conflict for the specific credential provider (in particular, assume role takes either `source_profile` or `credential_source` but not both), but not if there is configuration for multiple credential providers (that's why there's a hierarchy). It'd be like raising an error if you've specified direct credentials with `AWS_ACCESS_KEY_ID`, etc. and also had `AWS_DEFAULT_PROFILE` set. The AWS CLI also doesn't have this behavior.

Reply:
I've filed a bug against this as https://github.com/aws/aws-sdk-go/issues/3763. It's a deeper issue with the Go SDK's profile validation versus other SDKs, but should be fixed for this AWS SSO-related reason.